native/common/jp_context.cpp

```diff
@@ -246,6 +229,31 @@ void JPContext::startJVM(const string& vmPath, const StringVector& args,
 		m_ReferenceQueue->m_ReferenceQueue = JPObjectRef(frame,
 				frame.CallObjectMethodA(m_JavaContext.get(), getReferenceQueue, 0));
 
+		// Set up methods after everthing is start so we get better error
+		// messages
+		cls = m_ClassLoader->findClass(frame, "org.jpype.JPypeContext");
```

Comment:
There is an ever present miserable bug that that when something goes wrong in the bootstrapping process the exception handler will segfault.  Hopefully it is finally resolved, but I rearranged the order to make it harder to reoccur.